pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -827,6 +836,74 @@ allChainsLabel:
 	return mutable.Listener
 }
 
+func buildSidecarListenerTLSContext(tlsOpts *networking.Server_TLSOptions, sdsPath string, metadata *model.NodeMetadata) *auth.DownstreamTlsContext {
+	if tlsOpts == nil || (tlsOpts.Mode == networking.Server_TLSOptions_PASSTHROUGH) ||
+		(tlsOpts.Mode == networking.Server_TLSOptions_AUTO_PASSTHROUGH) {
+		return nil
+	}
+
+	tls := &auth.DownstreamTlsContext{
+		CommonTlsContext: &auth.CommonTlsContext{
+			AlpnProtocols: util.ALPNDownstream,
+		},
+	}
+
+	if tlsOpts.Mode == networking.Server_TLSOptions_MUTUAL ||
+		tlsOpts.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
```

Comment:
we should skip ISTIO_MUTUAL. This should be strictly simple/mutual/passthrough. But passthrough doesn't require any config. So just simple/mutual